src/pykiso/lib/auxiliaries/proxy_auxiliary.py

```diff
@@ -92,9 +92,42 @@ def __init__(
             is_proxy_capable=True, tx_task_on=False, rx_task_on=True, **kwargs
         )
         self.channel = com
+        self._open_count = 0
         self.logger = self._init_trace(activate_trace, trace_dir, trace_name)
         self.proxy_channels = self.get_proxy_con(aux_list)
 
+    @property
+    def _open_connections(self) -> int:
+        """A counter monitoring the number of attached running auxiliaries.
+
+        .. warning::
+            Do not set this manually as it can easily result in unexpected behaviours.
+
+
+        :getter: returns the number of currently attached and running auxiliaries.
+        :setter: set by ``CCProxy`` instances when being opened or closed.
+            When no open connection remains and the counter falls back to 0,
+            the ``ProxyAuxiliary`` will be stopped and the attached 'physical'
+            communication channel closed.
+            When the first connection is opened and the counter is increased
+            from 0 to 1, the ``ProxyAuxiliary`` will be started and the attached
+            'physical' communication channel opened.
+        """
+        with self.lock:
+            return self._open_count
+
+    @_open_connections.setter
+    def _open_connections(self, value: int):
```

Comment:
Would a counting-semaphore not do the trick?

Reply:
Checking when `proxy.open_connections.acquire(blocking=False)` returns False seems way more obscure to me than having an explicit counter and some boolean operations